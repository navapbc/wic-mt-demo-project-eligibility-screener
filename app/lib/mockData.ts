import cloneDeep from 'lodash/cloneDeep'

import type { SessionData } from '@src/types'
import { initialSessionData } from '@utils/sessionData'

// Helper functions to create valid data for each section of the session.
export function getMockEligibilityData() {
  return {
    residential: 'yes',
    categorical: ['pregnant', 'guardian'],
    previouslyEnrolled: 'no',
    adjunctive: ['fdpir', 'snap'],
  }
}

export function getMockChooseClinicData() {
  return {
    zipCode: '12345',
    clinic: {
      agency: 'AGENCY ZERO',
      agencyAddress: '0000 St, Helena, MT 00000',
      agencyTelephone: '(000) 000-0000',
      clinic: 'CLINIC ZERO',
      clinicAddress: '0000 St, Helena, MT 00000',
      clinicTelephone: '(000) 000-0000',
      county: 'COUNTY ZERO',
      id: 0,
      zip: '00000',
    },
  }
}

export function getMockIncomeData() {
  return {
    householdSize: '3',
  }
}

export function getMockContactData() {
  return {
    firstName: 'Jack',
    lastName: 'O Lantern',
    phone: '1231231234',
    comments: 'comments',
  }
}

// Helper function to setup a fully filled out session.
export function getMockSessionData() {
  const mockSession = getEmptyMockSession()
  mockSession.eligibility = getMockEligibilityData()
  mockSession.chooseClinic = getMockChooseClinicData()
  mockSession.contact = getMockContactData()
  return mockSession
}

// Helper function te setup an empty session that passes typechecking.
export function getEmptyMockSession(): SessionData {
  return cloneDeep(initialSessionData)
}
