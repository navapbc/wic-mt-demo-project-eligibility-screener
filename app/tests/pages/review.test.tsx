import { render, screen } from '@testing-library/react'

import Review from '@pages/review'

import { getMockSessionData } from '../helpers/mockData'
import { setup } from '../helpers/setup'
import {
  testAccessibility,
  testActionButtonRoute,
  testBackLink,
  testSnapshot,
} from '../helpers/sharedTests'

/**
 * Test setup
 */

const route = '/review'
const backRoute = '/contact'
const forwardRoute = '/confirmation'

/**
 * Begin tests
 */

it('should match full page snapshot', () => {
  setup(route)
  const mockSession = getMockSessionData()
  testSnapshot(
    <Review
      session={mockSession}
      backRoute={backRoute}
      forwardRoute={forwardRoute}
    />
  )
})

it('should pass accessibility scan', async () => {
  const { mockSession } = setup(route)
  await testAccessibility(
    <Review
      session={mockSession}
      backRoute={backRoute}
      forwardRoute={forwardRoute}
    />
  )
})

it('should have a back link that matches the backRoute', () => {
  const { mockSession } = setup(route)
  testBackLink(
    <Review
      session={mockSession}
      backRoute={backRoute}
      forwardRoute={forwardRoute}
    />,
    backRoute
  )
})

it('should have an action button that routes to /confirmation', async () => {
  const { mockSession, user } = setup(route)
  const element = (
    <Review
      session={mockSession}
      backRoute={backRoute}
      forwardRoute={forwardRoute}
    />
  )
  testActionButtonRoute(element, '/confirmation', 'Submit', user)
})
