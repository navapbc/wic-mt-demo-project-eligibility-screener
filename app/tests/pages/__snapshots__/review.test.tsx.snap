// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match full page snapshot 1`] = `
<div>
  <a
    class="usa-link"
    href="/contact"
  >
    Back
  </a>
  <h1>
    Review and submit your information
  </h1>
  <p>
    Review the information on this page before submitting. If you want to make changes, select "Edit" to go to the section you want to change.
  </p>
  <div
    class="review-collection margin-top-6 border-bottom-1px"
  >
    <h2>
      Eligibility questions
      <div
        class="float-right"
      >
        <button
          class="usa-button usa-button--small display-block usa-button--unstyled margin-top-1"
        >
          Edit
        </button>
      </div>
    </h2>
    <dl
      class="margin-bottom-2"
    >
      <div
        class="review-element margin-bottom-3"
      >
        <dt>
          <strong>
            Do you live or work in Montana?
          </strong>
        </dt>
        <dd
          class="margin-left-0"
        >
          Yes
        </dd>
      </div>
      <div
        class="review-element margin-bottom-3"
      >
        <dt>
          <strong>
            Please select all the following that apply to your household:
          </strong>
        </dt>
        <dd
          class="margin-left-0"
        >
          <ul>
            <li>
              I'm pregnant
            </li>
            <li>
              I'm the guardian or foster parent of an infant or child under the age of 5 years old
            </li>
          </ul>
        </dd>
      </div>
      <div
        class="review-element margin-bottom-3"
      >
        <dt>
          <strong>
            Have you or someone in your household received WIC benefits before?
          </strong>
        </dt>
        <dd
          class="margin-left-0"
        >
          No
        </dd>
      </div>
      <div
        class="review-element margin-bottom-3"
      >
        <dt>
          <strong>
            Are you or someone in your household currently enrolled in any of the following programs in Montana?
          </strong>
        </dt>
        <dd
          class="margin-left-0"
        >
          <ul>
            <li>
              FDPIR (Food Distribution Program on Indian Reservations)
            </li>
            <li>
              SNAP (Supplemental Nutrition Assistance Program)
            </li>
          </ul>
        </dd>
      </div>
    </dl>
  </div>
  <div
    class="review-collection margin-top-6 border-bottom-1px"
  >
    <h2>
      Choose a clinic
      <div
        class="float-right"
      >
        <button
          class="usa-button usa-button--small display-block usa-button--unstyled margin-top-1"
        >
          Edit
        </button>
      </div>
    </h2>
    <dl
      class="margin-bottom-2"
    >
      <div
        class="review-element margin-bottom-3"
      >
        <dt>
          <strong>
            Residential ZIP code
          </strong>
        </dt>
        <dd
          class="margin-left-0"
        >
          <div>
            12345
          </div>
        </dd>
      </div>
      <div
        class="review-element margin-bottom-3"
      >
        <dt>
          <strong>
            You selected this WIC Clinic to visit
          </strong>
        </dt>
        <dd
          class="margin-left-0"
        >
          <div
            class="clinic-info"
          >
            <div
              class="clinic-name"
            >
              CLINIC ZERO
            </div>
            <div
              class=""
            >
              <div
                class="clinic-street-address"
              >
                0000 St, Helena, MT 00000
              </div>
              <div
                class="clinic-phone"
              >
                (000) 000-0000
              </div>
            </div>
          </div>
        </dd>
      </div>
    </dl>
  </div>
  <div
    class="review-collection margin-top-6 border-bottom-1px"
  >
    <h2>
      Contact information
      <div
        class="float-right"
      >
        <button
          class="usa-button usa-button--small display-block usa-button--unstyled margin-top-1"
        >
          Edit
        </button>
      </div>
    </h2>
    <dl
      class="margin-bottom-2"
    >
      <div
        class="review-element margin-bottom-3"
      >
        <dt>
          <strong>
            First name
          </strong>
        </dt>
        <dd
          class="margin-left-0"
        >
          Jack
        </dd>
      </div>
      <div
        class="review-element margin-bottom-3"
      >
        <dt>
          <strong>
            Last name
          </strong>
        </dt>
        <dd
          class="margin-left-0"
        >
          O Lantern
        </dd>
      </div>
      <div
        class="review-element margin-bottom-3"
      >
        <dt>
          <strong>
            Phone number
          </strong>
        </dt>
        <dd
          class="margin-left-0"
        >
          123-123-1234
        </dd>
      </div>
      <div
        class="review-element margin-bottom-3"
      >
        <dt>
          <strong>
            Comments, questions, other information
          </strong>
        </dt>
        <dd
          class="margin-left-0"
        >
          comments
        </dd>
      </div>
    </dl>
  </div>
  <button
    class="usa-button usa-button--small display-block margin-top-6"
  >
    Submit
  </button>
</div>
`;
